### ISSUE TITLE
tabBarVisible: false does not work on a stackNavigation child

### ISSUE THREAD

**Issue by alebupal** [react-navigation/react-navigation#5790]
I have a TabNavigator, inside this I have redirections to children, which are StackNavigator. The problem is that within the children, I can not hide the tabNavigator with tabBarVisible: false

THE PARENT (TabNavigator)
```
import React from 'react';
import { Text } from 'react-native';
import { createBottomTabNavigator, createStackNavigator, createAppContainer } from 'react-navigation';
import { Home, Search, Add, Follow, Profile } from '../screens';
import HomeNavigator from './Home';
import SearchNavigator from './Search';

const config = {
    headerMode: 'none',
    tabBarPosition: 'top'
};

const screens = {
    Home: {
        screen: HomeNavigator,
        navigationOptions: ({navigation}) => ({
            title: 'Home',
        })
    }
};
```

const Authenticated = createBottomTabNavigator(screens, config);

export default createAppContainer(Authenticated);

THE CHILDREN (StackNavigator)

```
import React from 'react';
import { createStackNavigator, createAppContainer } from 'react-navigation';
import { Home, Profile } from '../screens';
import { Posts, Comments } from '../screens/screensHome';

const screens = {
    Home: {
        screen: Home,
        navigationOptions: {
            header: null,
        }
    },
    Comments: {
        screen: Comments
    }
};

const HomeNavigation = createStackNavigator(screens, config);

export default createAppContainer(HomeNavigation);
```

THE VIEW (width navigationOptions)

```
import React, { Component } from 'react';
import {
  View,
  Text,
  StyleSheet,
} from 'react-native';

export default class Comments extends Component {

    static navigationOptions = {
        tabBarVisible: false
    }

    render() {
        return (
            <View>
                <Text>I'm the Comments component</Text>
            </View>
        );
    }
}
```

I leave here the code of the app 

[https://snack.expo.io/@ricarquico/clone-instagram][1]

  [1]: https://snack.expo.io/@ricarquico/clone-instagram

in short I want to hide this http://prntscr.com/n6hw68

| software         | version
| ----------------    | -------
| react-navigation | 3.6.0
| react-native        | 32.0.0

**Comment by brentvatne** [comment]
https://reactnavigation.org/docs/en/navigation-options-resolution.html

**Comment by 2diegoduque** [comment]
The only thing that worked for me
```
const homeScreenStack = createStackNavigator({
  Home: {
    screen: Home,
    navigationOptions: ({ navigation }) => ({
      title: "Home"
    })
  },
  Comments: {
    screen: Comments,
    navigationOptions: ({ navigation }) => ({
      title: "Comments"
    })
  }
});

homeScreenStack.navigationOptions = ({ navigation }) => {
  let tabBarVisible = true;
  for (let i = 0; i < navigation.state.routes.length; i++) {
    if (navigation.state.routes[i].routeName == "Comments") {
      tabBarVisible = false;
    }
  }

  return {
    tabBarVisible
  };
};
```

**Comment by vickkeshri** [comment]
It is working but the header height of stack navigation is increase, which makes the screens looks alike.

**Comment by DadaConsultant** [comment]
Based on **2diegoduque**  solution above. 
I modified as below.

let tabBarVisible = true;

  let routes = navigation.state.routes;
  if(routes[routes.length-1].routeName == "xxx" || 
      routes[routes.length-1].routeName == "xxx" || 
      routes[routes.length-1].routeName == "xxxx"){
      tabBarVisible = false;
  }

  return {
    tabBarVisible
  };

**Comment by qianlanse** [comment]
V5 view https://www.youtube.com/watch?v=sH6OqX6ANeE

**Comment by c8p** [comment]
https://reactnavigation.org/docs/4.x/navigation-options-resolution/

`const FeedStack = createStackNavigator({
  FeedHome: FeedScreen,
  Details: DetailsScreen,
});

FeedStack.navigationOptions = ({ navigation }) => {
  let tabBarVisible = true;
  if (navigation.state.index > 0) {
    tabBarVisible = false;
  }

  return {
    tabBarVisible,
  };
};`

**Comment by ghasemikasra39** [comment]
same here:
```
    "@react-navigation/bottom-tabs": "^5.3.1",
    "@react-navigation/native": "^5.2.1",
    "@react-navigation/stack": "^5.2.16",
```

**Comment by Menganitou** [comment]
https://reactnavigation.org/docs/hiding-tabbar-in-screens/

**Comment by emersonjds** [comment]
Awesome resolution @2diegoduque
